subsys/net/ip/ipv6_nbr.c

```diff
@@ -664,8 +664,13 @@ enum net_verdict net_ipv6_prepare_for_send(struct net_pkt *pkt)
 	 */
 	if (net_pkt_ipv6_fragment_id(pkt) == 0) {
 		size_t pkt_len = net_pkt_get_len(pkt);
+		u16_t mtu = NET_IPV6_MTU;
```

Comment:
Alternatively, we could also do something like this:
```
u16_t mtu = NET_IPV6_MTU;
u16_t iface_mtu = net_if_get_mtu(net_pkt_iface(pkt));

mtu = MAX(mtu, iface_mtu);
if (mtu > pkt_len) {
```

Reply:
yep using MAX would simplify, no need of intermediate variable as iface_mtu I guess though

Reply:
I think it would be better to have an intermediate variable, reason being that the MAX() is a macro in which case it might end up calling `net_if_get_mtu(net_pkt_iface(pkt))` twice.

Reply:
so? if I declare mtu = net_if_get_mtu(net_pkt_iface(pkt));
and then: mtu = max(mtu, NET_IPV6_MTU); 
It will be totally fine there.

Note also that net_if_get_mtu() is an inline function, accessing sturct attributes, so even called twice in the code it's most likely not being an issue (compiler would opmitize it out)

Reply:
Yes, probably compiler will be able to optimize extra calls away. I just find it bad habit to have function calls (regardless if they are inlined or not) in macros, eventually it will bite you. Been there, seen that :)